api/types/client.go

```diff
@@ -285,10 +285,19 @@ type ServiceCreateOptions struct {
 }
 
 // ServiceCreateResponse contains the information returned to a client
-// on the  creation of a new service.
+// on the creation of a new service.
 type ServiceCreateResponse struct {
 	// ID is the ID of the created service.
 	ID string
+	// Warning is a non-fatal warning message to pass on to the user.
+	Warning string `json:",omitempty"`
```

Comment:
Not in favor of a boolean because it's important to communicate the actual error message. I think a general string field is necessary. However, we could add a boolean in addition to indicate whether pinning by digest happened if people think it's a good idea.